### PR TITLE
(PC-14998) feat(phoneVal): send validation code to backend

### PR DESCRIPTION
**Pull request by yorickeando**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-14998

## Checklist

I have:

- [ ] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** or **GitHub dev name** for any added TODO / FIXME.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)

## Screenshots

Capture d’écran 2022-06-30 à 16 52 08

https://user-images.githubusercontent.com/44037911/176913353-2e4e3316-b02d-4bd8-ae20-88cc645f5281.mov

https://user-images.githubusercontent.com/44037911/176913368-830dcf0f-8ae0-433d-bd85-b103b03cbd4f.mov

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/features/identityCheck/pages/phoneValidation/SetPhoneValidationCode.tsx [hunk 1]
```diff
@@ -24,17 +32,61 @@ export type SetPhoneValidationCodeProps = StackScreenProps<
...
+    } else {
+      setErrorMessage(extractApiErrorMessage(err))
+    }
+  }
+
+  const { mutate: validatePhoneNumber, isLoading } = useValidatePhoneNumberMutation({
+    onSuccess: () => {
+      navigate('IdentityCheckStepper')
+    },
+    onError: onValidatePhoneNumberError,
```
Comment (anoukhello) [inline src/features/identityCheck/pages/phoneValidation/SetPhoneValidationCode.tsx:68]: Ça te dirait de mettre le contenu de onValidatePhoneNumberError directement ici ?

Comment (yorickeando) [reply src/features/identityCheck/pages/phoneValidation/SetPhoneValidationCode.tsx:68]: [I accept](https://tenor.com/view/edna-mode-push-too-hard-accept-yes-insist-gif-13750631.gif)

Comment (yorickeando) [reply src/features/identityCheck/pages/phoneValidation/SetPhoneValidationCode.tsx:68]: I accept

File: src/features/identityCheck/pages/phoneValidation/__tests__/SetPhoneValidationCode.native.test.tsx [hunk 2]
```diff
@@ -30,10 +39,74 @@ describe('SetPhoneValidationCode', () => {
...
+
+    await waitFor(() => {
+      expect(navigate).toHaveBeenCalledWith('IdentityCheckStepper')
+    })
+  })
 })
 
-function renderSetPhoneValidationCode() {
-  return render(<SetPhoneValidationCode {...navigationProps} />, {
+function renderSetPhoneValidationCode(props?: Partial<SetPhoneValidationCodeProps>) {
```
Comment (anoukhello) [inline src/features/identityCheck/pages/phoneValidation/__tests__/SetPhoneValidationCode.native.test.tsx:108]: Au fait tu utilises qlq part ces props optionnelles ?

**Pull request merged**